File: Lab1/Homework/Homework/Num3.cs
using System;

namespace Homework
{
    class Num3
    {
        //для невпорядкованого масиву
        int CountUnorderedArray<T>(T[] arr)
        {
            List<T> uniqueElements = new List<T>();
            foreach (var element in arr)
            {
                if (!uniqueElements.Contains(element))
                {
                    uniqueElements.Add(element);
                }
            }
            return uniqueElements.Count;
        }

        //для впорядкованого масиву
        int CountOrderedArray<T>(T[] arr)
        {
            int uniqueCount = 1; // Початково в масиві точно є один унікальний елемент
            for (int i = 1; i < arr.Length; i++)
            {
                if (arr[i] != arr[i - 1])
                {
                    uniqueCount++;
                }
            }
            return uniqueCount;
        }

        static void Main(string[] args)
        {
            
        }
    }
}

File: Homework2/Num1/num3.py
def func(arr, x):
    i, j = 0, 0
    while i < len(arr[0]) or j < len(arr[1]):
        if arr[0][i] == x or arr[1][j] == x:
            return True
        else:
            i += 1
            j += 1
    return False


arr = [[2, 4, 1, 6, 7], [2, 7, 8, 12, 4]]
print(func(arr, 4))
